### ISSUE TITLE
Glass Model not populated in GlassView after upgrade

### ISSUE THREAD

**Issue by tomunderhill** [mikeedwards83/Glass.Mapper#125]
We recently tried to upgrade to the latest Glass Nuget packages and had the following problem...

A view rendering using a GlassView no longer populated the model correctly (all the properties were null). 

The exact same rendering works fine when moved to a placeholder defined on more base-level rendering, but no longer works when in a placeholder defined in a GlassView nested a bit more in the page rendering set-up.

However, if I use GlassHtml.SitecoreContext.GetCurrentItem within the rendering, the item resolves correctly.

**Comment by mikeedwards83** [comment]
Can you give me a screenshot of the the rendering setup in both scenarios?

**Comment by tomunderhill** [comment]
![image](https://cloud.githubusercontent.com/assets/476379/7374204/c58f81d2-edc8-11e4-99c9-2ffe8a57a5d0.png)

This is the rendering setup - on the highlighted item, if you change the placeholder it is in from 'Accordion Pane 1' to 'Body', it works.

**Comment by tomunderhill** [comment]
More detail...

The accordion rendering, in which the broken Course Pane Overview should sit:

```
@inherits Glass.Mapper.Sc.Web.Mvc.GlassView<SectionTitleContainerModel>

@if (Model != null)
{
    var panesItems = Model.Children.OrderBy(x => x.SortOrder).ToList();

    <section class="accordion--course-details accordion">
        @for (int i = 0; i < panesItems.Count; i++)
{
    <div class="accordion__section @(i == 0 ? "highlighted-item" : null)" id="@panesItems[i].SectionTitle.ToLowerAlphanumeric()">
        <h3 class="accordion--course-details__heading accordion__heading">@panesItems[i].SectionTitle</h3>
        <div class="accordion__content">
            @Html.Sitecore().Placeholder("Accordion Pane " + (i + 1))
        </div>
    </div>
}
    </section>
}
```

The model here, SectionTitleContainerModel, is unrelated to the model of the 'page' Sitecore item

**Comment by cardinal252** [comment]
Call me crazy - but is the entry still present in the mvc.GetModel pipeline and if so - what is it ?

**Comment by tomunderhill** [comment]
```
<mvc.getModel>
        <processor type="Glass.Mapper.Sc.Pipelines.Response.GetModel, Glass.Mapper.Sc.Mvc"/>
      </mvc.getModel>
```

You're crazy, but don't call me Shirley.

**Comment by cardinal252** [comment]
I assume you are seeing that in the appropriate showconfig.aspx ??

What version of Glass.Mapper.Sc.Mvc?

**Comment by cardinal252** [comment]
Also - in the correct language ?

**Comment by tomunderhill** [comment]
Upgrading from:

```
 <package id="Glass.Mapper" version="3.0.10.23" targetFramework="net451" />
  <package id="Glass.Mapper.Sc" version="3.2.0.34" targetFramework="net451" />
  <package id="Glass.Mapper.Sc.CastleWindsor" version="3.2.0.17" targetFramework="net451" />
  <package id="Glass.Mapper.Sc.Mvc-5" version="3.2.0.1" targetFramework="net451" />
``` 

to

```
<package id="Glass.Mapper" version="3.0.15.32" targetFramework="net451" />
  <package id="Glass.Mapper.Sc" version="3.3.1.53" targetFramework="net451" />
  <package id="Glass.Mapper.Sc.CastleWindsor" version="3.3.1.26" targetFramework="net451" />
  <package id="Glass.Mapper.Sc.Mvc-5" version="3.3.1.48" targetFramework="net451" />
```

It's in showconfig ok. As I mentioned, it is working fine when placeholder changes, which is odd. Also, there are other Glass view renderings on the page working ok.

**Comment by cardinal252** [comment]
What does the model look like ?

**Comment by tomunderhill** [comment]
```
[SitecoreType(AutoMap = true, TemplateId = ContentReferences.Course.CourseBaseTemplateId)]
public class CourseBaseModel : PageModel, CourseAvailabilityEnabledModel, ICourseSettingsModel, ICampaignCourseModel
{
        [IndexField("alphapagetitlesort")]
        public string PageTitleSort { get; set; }

        [IndexField("Introduction")]
        public virtual string Introduction { get; set; }

        [IndexField("Overview")]
        public virtual string Overview { get; set; }

        [IndexField("Careers")]
        public virtual string Careers { get; set; }

        [SitecoreQuery(ContentReferences.SupportingItemPath + "/Settings")]
        public virtual GlobalSettingsModel GlobalSettings { get; set; }

        [SitecoreIgnore]
        public string OverviewWithDefaultShowHideText
        {
            get
            {
                if (Overview == null)
                {
                    return string.Empty;
                }
                return this.Overview.Replace("{LinkText}", this.GlobalSettings.ShowHideLinkText);
            }
        }
        
        public virtual string WhereYouStudy { get; set; }
        
        [IgnoreIndexField]
        public virtual LookupReferenceModel LevelOfStudy { get; set; }

        [IgnoreIndexField]
        public virtual FacultyModel Faculty { get; set; }

        [IgnoreIndexField]
        public virtual DepartmentModel Department { get; set; }

        [IgnoreIndexField]
        [SitecoreField(Setting = SitecoreFieldSettings.InferType)]
        public virtual IEnumerable<LookupReferenceModel> ModeOfStudy { get; set; }

        [IgnoreIndexField]
        public virtual IEnumerable<LookupReferenceModel> Availability { get; set; }

        [IgnoreIndexField]
        public virtual IEnumerable<LookupReferenceModel> Categories { get; set; }

        [IgnoreIndexField]
        public virtual IEnumerable<LookupReferenceModel> Duration { get; set; }

        [IgnoreIndexField]
        public virtual IEnumerable<LookupReferenceModel> StartDate { get; set; }
        
        public virtual string SitsCode { get; set; }
        
        public virtual string TeachingTimes { get; set; }

        public virtual string AdditionalStudyInfo { get; set; }

        //this field has display title Campus in Sitecore
        [IgnoreIndexField]
        public virtual IEnumerable<CampusModel> Location { get; set; }

        [IgnoreIndexField]
        public virtual IEnumerable<Guid> LocationOverride { get; set; }

        [IgnoreIndexField]
        public virtual IEnumerable<RichTextModel> EntryRequirementInfo { get; set; }

        [IgnoreIndexField]
        public virtual IEnumerable<RichTextModel> HighlightedEntryInfo { get; set; }

        [IgnoreIndexField]
        public virtual IEnumerable<CallToActionModel> CallToAction { get; set; }

        [IgnoreIndexField]
        [SitecoreQuery(ContentReferences.Course.SupportingPath)]
        public virtual CourseSettingsModel CourseSettings { get; set; }

        [IgnoreIndexField]
        [SitecoreField("__Renderings")]
        public virtual string Renderings { get; set; }

        private IList<CallToActionModel> callToActionList;

        [SitecoreIgnore]
        public virtual IList<CallToActionModel> CallToActionList
        {
            get
            {
                if (this.callToActionList == null)
                    this.callToActionList = this.CallToAction.ToList();
                return this.callToActionList;
            }
        }
      
    }
```

**Comment by mikeedwards83** [comment]
Are you using a DataSource for the rendering?

**Comment by tomunderhill** [comment]
No, it's picking up the 'page' item

**Comment by mikeedwards83** [comment]
Does this happen for all renderings placed at that level or is it just this one particular rendering? What happens if you move the rendering to a deeper nested placeholder?

**Comment by behrreich** [comment]
I'm having the same problem with a View Rendering.  If I inherit from

Glass.Mapper.Sc.Web.Mvc.GlassView<Sitecore.Mvc.Presentation.RenderingModel>

I can retrieve the Model and do a GlassCast to my custom type

var item = Model.Item.GlassCast<ThumbnailGallery>();

However, If I inherit from

Glass.Mapper.Sc.Web.Mvc.GlassView<ThumbnailGallery>

my Model is always empty with item id of zero.  Below is the relevant showconfig entry, which looks fine to me.  In this case we are using Glass.Mapper.dll version 4.0.1.5

Thanks for your help.

Bill E.

<mvc.getModel patch:source="Glass.Mapper.Sc.config">
<processor type="Glass.Mapper.Sc.Pipelines.Response.GetModel, Glass.Mapper.Sc.Mvc"/>
<processor type="Glass.Mapper.Sc.Pipelines.Response.GetModelFromView, Glass.Mapper.Sc.Mvc"/>
<processor type="Sitecore.Mvc.Pipelines.Response.GetModel.GetFromItem, Sitecore.Mvc" patch:source="Sitecore.Mvc.config"/>
<processor type="Sitecore.Mvc.Pipelines.Response.GetModel.GetFromProperty, Sitecore.Mvc" patch:source="Sitecore.Mvc.config"/>
<processor type="Sitecore.Mvc.Pipelines.Response.GetModel.GetFromLayout, Sitecore.Mvc" patch:source="Sitecore.Mvc.config"/>
<processor type="Sitecore.Mvc.Pipelines.Response.GetModel.GetFromRenderingItem, Sitecore.Mvc" patch:source="Sitecore.Mvc.config"/>
<processor type="Sitecore.Mvc.Pipelines.Response.GetModel.CreateDefaultRenderingModel, Sitecore.Mvc" patch:source="Sitecore.Mvc.config"/>
<processor type="Sitecore.Mvc.Pipelines.Response.GetModel.InitializeModel, Sitecore.Mvc" patch:source="Sitecore.Mvc.config"/>
</mvc.getModel>

**Comment by mikeedwards83** [comment]
Hi Bill 

Can you post the two version of the cshtml file you are using?

Mike

**Comment by behrreich** [comment]
Mike,

The strangest thing happened.  I was using the master database for testing.  However, I did a publish, switched to the web database and it started working!  It is also working on master now.  Something else I noticed is that I am not specifying the model in my rendering but it seems to GlassCast to the correct type anyway.  I haven't seen this before.

I don't know what happened to make it start working.  I'm quite confused.  I have attached a screenshot of the code for my two renderings.

![image](https://cloud.githubusercontent.com/assets/945090/9004579/23dbb6c8-3745-11e5-8904-2ac29400799d.png)
